src/blockdata/transaction.rs

```diff
@@ -220,12 +220,104 @@ impl Default for TxIn {
         TxIn {
             previous_output: OutPoint::default(),
             script_sig: Script::new(),
-            sequence: u32::max_value(),
+            sequence: Sequence::MAX,
             witness: Witness::default(),
         }
     }
 }
 
+/// Bitcoin transaction input sequencing number.
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct Sequence(pub u32);
+
+impl Sequence {
+    /// The maximum allowable sequence number
+    pub const MAX: Self = Sequence(MAX_SEQUENCE);
+    /// The lowest sequence number that does not opt-in for replace-by-fee.
+    pub const RBF_SIGNAL: Self = Sequence(0xFFFFFFFE);
+    /// Zero value sequence
+    pub const ZERO: Self = Sequence(0);
+    /// BIP-68 relative time lock disable flag mask
+    const DISABLE_FLAG_MASK: u32 = 0x80000000;
+    /// BIP-68 relative time lock type flag mask
+    const LOCK_TYPE_MASK: u32 = 0x00400000;
+    
+    /// Retuns `true` if the sequence number is 0xffffffff.
+    #[inline]
+    pub fn is_final(&self) -> bool {
+        *self == Sequence::MAX
+    }
+
+    /// Returns `true` if the sequence number is less than 0xfffffffe which
+    /// indicates that the transaction is opted-in to BIP125 replace-by-fee.
+    #[inline]
+    pub fn is_rbf(&self) -> bool {
+        *self < Sequence::RBF_SIGNAL
+    }
+
+    /// Returns `true` if the sequence has a relative locktime.
+    #[inline]
+    pub fn is_relative_lock_time(&self) -> bool {
+        self.0 & Sequence::DISABLE_FLAG_MASK == 0
+    }
+
+    /// Returns `true` if the sequence number encodes a block based relative lock-time.
+    #[inline]
+    pub fn is_height_locked(&self) -> bool {
+        self.is_relative_lock_time() & (self.0 & Sequence::LOCK_TYPE_MASK == 0)
+    }
+
+    /// Returns `true` if the sequene number encodes a time interval based relative lock-time.
+    #[inline]
+    pub fn is_time_locked(&self) -> bool {
+        self.is_relative_lock_time() & (self.0 & Sequence::LOCK_TYPE_MASK > 0)
+    }
+    
+    /// Create a relative lock-time using block height.
+    #[inline]
+    pub fn from_height(height: u16) -> Self {
+        Sequence(u32::from(height))
+    }
+
+    /// Create a relative lock-time using time units where each unit is equivalent to 512 seconds.
+    #[inline]
+    pub fn from_timeunits(units: u16) -> Self {
```

Comment:
I've gone ahead and added `from_seconds_floor` and `from_seconds_ceiling` but have them returning `Result<Self, RelativeLockTimeError>` due to the max possible seconds being larger that u16 and conversion from u32 to u16 being fallible.

Reply:
Also, if I am to add the `RelativeTimeLock` enum in this PR, these methods would be implemented on the enum instead of `Sequence` and `Sequence` will instead have `pub fn from_lock_time(lock_time: RelativeLockTime) -> Sequence`

Reply:
a